extensions/amp-carousel/0.1/base-slides.js

```diff
@@ -147,6 +154,25 @@ export class BaseSlides extends BaseCarousel {
         this.autoplayDelay_);
   }
 
+  /**
+   * Called on user interaction to toggle the autoplay feature.
+   */
+  toggleAutoPlay(toggleOn) {
+    // Set the autoPlay status to the opposite
```

Comment:
few suggestions here:
1- if `toggleOn` equals previous value of `this.shouldAutoplay_` this should be a noop so we can bail out early.
2- I think `setupAutoplay_()` needs to be called if we are going from no-autoplay to autoplay for the first time.
3- I was hoping the rest can become as simple as 
```
if(this.shouldAutoplay_) {
this.autoplay_();
} else {
this.clearAutoplay_();
}
``` 
similar to how `onViewportCallback` handles pausing and resuming autoplay as carousel is moved in.out of the viewport.

Reply:
1- Done
2 - Done
3 - Done